### PR TITLE
PWA Shell with service worker

### PR DESCRIPTION
**Pull request by dvoytenko**

### CODE REVIEW

File: examples/pwa-sw.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+});
+
+self.addEventListener('fetch', event => {
+  // TODO(dvoytenko): use cache, implement one-behind.
+  if (event.request.url.indexOf('amp.max.html') != -1) {
+    // Override response with the shell unless the leaf document is explicitly
+    // requested.
+    const isDirectFetch = event.request.headers.get('AMP-Direct-Fetch') == '1';
+    if (!isDirectFetch) {
+      event.respondWith(fetch('/pwa'));
```
Comment (cramforce) [inline examples/pwa-sw.js]: Don't you still, even without the cache, want to initiate the request here? The browser cache should do the right thing.

File: examples/pwa.js [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+   * @return {string}
+   */
+  getOrigin_(relativeUrlString) {
+    return new URL(relativeUrlString, this.baseUrl_).origin;
+  }
+
+  /**
+   */
+  onMessage_(type, data, rsvp) {
+    log('receieved message:', type, data, rsvp);
```
Comment (cramforce) [inline examples/pwa.js]: nit: typo

File: examples/pwa.html [hunk 3]
```diff
@@ -0,0 +1,111 @@
+<!doctype html>
```
Comment (cramforce) [inline examples/pwa.html]: Please create a sub directory

File: examples/pwa-sw.js (hunk 1 above)
Comment (dvoytenko) [inline examples/pwa-sw.js]: Yes. Definitely! Done.

File: examples/pwa.js (hunk 2 above)
Comment (dvoytenko) [inline examples/pwa.js]: better sure than sorry :)

File: examples/pwa.html (hunk 3 above)
Comment (dvoytenko) [inline examples/pwa.html]: done

**Comment by dvoytenko** [comment]
@cramforce PTAL

**Comment by cramforce** [comment]
Got distracted. Commit at will. LGTM

**Pull request merged**